### PR TITLE
houdini: 17.5.327 -> 18.0.391

### PR DESCRIPTION
**Pull request by kwohlfahrt**
##### Motivation for this change

Upstream package is no longer available for download, also new version is released.

Due to the first part (`17.5.327` no longer available for download) at least the first commit should be backported to stable I think?

I'm happy to add myself as a maintainer if @canndrew has no objections?

@emptyflask and @PlumpMath were working on the v18.0 update, so they might also be interested in this PR.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emptyflask** [comment]
Thanks for doing this, I was trying to install v18 but kept running into issues with qt5.

I just tested this commit with `version = "18.0.460";` and `sha256 = "18rbwszcks2zfn9zbax62rxmq50z9mc3h39b13jpd39qjqdd3jsd";` and the program started up.

We should maybe replace the message with something that tells the user to use `nix-prefetch-url file:///path/to/houdini-version-linux_x86_64_gcc6.3.tar.gz` since they probably won't be using the URL to SideFX's CDN.

**Comment by kwohlfahrt** [comment]
@emptyflask that hash is good for the updated version, so I've bumped it again. This PR removes the custom message for `requireFile`, so we end up with the default:

```
  Unfortunately, we cannot download file houdini-18.0.460-linux_x86_64_gcc6.3.tar.gz automatically.
  Please go to https://www.sidefx.com to download it yourself, and add it to the Nix store
  using either
    nix-store --add-fixed sha256 houdini-18.0.460-linux_x86_64_gcc6.3.tar.gz
  or
    nix-prefetch-url --type sha256 file:///path/to/houdini-18.0.460-linux_x86_64_gcc6.3.tar.gz
```

Does that seem suitable? It suggests your option and also an alternative using `nix-store`.

**Comment by emptyflask** [comment]
Looks good to me

**Comment by ryantm** [comment]
There's been no response from the maintainer @canndrew  so if either of you want to add yourself as maintainer, then I'd be okay merging it.

**Comment by emptyflask** [comment]
I don't mind being the maintainer -- but if someone else who actually uses Houdini on a daily basis would rather do it, that would make more sense.

BTW, I've attempted to upgrade this to v18.5 after modifying the license date, but it looks like SideFX changed the license server in their latest release somehow, so I haven't been able to start it up yet. I'm not sure if I'll have the time to dedicate to that amount of work needed to get that running, but I'll give it a shot.

**Comment by arximboldi** [comment]
I've found this on the Houdini website: ftp://ftp.sidefx.com/public

Maybe it can be used to automatically download the software instead of requiring `nix-prefetch-url`?

**Comment by kwohlfahrt** [comment]
@ryantm sorry, I missed that you wanted one of us as maintainer before merging. I've added myself now, and rebased onto latest master. I also added the ability to override the unwrapped package, as I did in #124906. This will make it easier to do a minor version bump in a personal overlay.

I'm currently working on v18.5, and will add that patch to this PR if it has not been merged by the time I figure it out. Otherwise, I will create a new PR.

@arximboldi - that looks great. I can see the mention of it [here](www.sidefx.com/faq/question/where-do-i-download-houdini-from/), so it seems like SideFX are OK with using FTP to get packages.

**Comment by kwohlfahrt** [comment]
@arximboldi - I'm actually finding using the FTP server with `fetchurl` to be a bit flaky (sometimes it times out), so I'll leave it as-is for this PR, but might revisit that later after some more testing.

File: pkgs/applications/misc/houdini/runtime.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
-{ lib, stdenv, requireFile, zlib, libpng, libSM, libICE, fontconfig, xorg, libGLU, libGL, alsa-lib, dbus, xkeyboardconfig, bc, addOpenGLRunpath }:
+{ lib, stdenv, requireFile, zlib, libpng, libSM, libICE, fontconfig, xorg, libGLU, libGL, alsa-lib, dbus, xkeyboardconfig, nss, nspr, expat, pciutils, libxkbcommon, bc, addOpenGLRunpath }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/houdini/runtime.nix:1]:
```suggestion
{ lib, stdenv, requireFile, zlib, libpng, libSM, libICE, fontconfig, xorg, libGLU, libGL, alsa-lib
, dbus, xkeyboardconfig, nss, nspr, expat, pciutils, libxkbcommon, bc, addOpenGLRunpath
}:
```

Comment (kwohlfahrt) [reply pkgs/applications/misc/houdini/runtime.nix]: Committed.

**Comment by kwohlfahrt** [comment]
I have a working derivation for 18.5 now, but I'd like to merge this PR separately as it's a significant rework, and there is a slight regression - `sesinetd` must be started manually on 18.5.

**Pull request merged**